vllm/core/block_manager.py

```diff
@@ -290,8 +301,9 @@ def append_slot(
         if last_block.ref_count == 1:
             # Not shared with other sequences. Appendable.
             # If the last block is now complete, promote it to a full block so that it can be shared
-            new_block = self._maybe_promote_last_block(seq, last_block)
-            block_table[-1] = new_block
+            if self.enable_caching:
```

Comment:
Nit: the same check is in _maybe_promote_last_block. Let's remove it here for simplicity.

Reply:
I removed it from `_maybe_promote_last_block()` because adding the block to the end of block table also depends on this condition.